flake.nix

```diff
@@ -1,27 +1,44 @@
 {
   description = "nix-index database";
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
 
-  outputs = { self, nixpkgs, ... }:
+  outputs =
+    { self, nixpkgs, ... }:
     let
       inherit (nixpkgs) lib;
 
-      systems = lib.attrNames self.legacyPackages;
-      testSystems = [ "x86_64-linux" "aarch64-linux" ];
-
-      databases = import ./packages.nix;
-
-      mkPackages = pkgs: {
-        nix-index-with-db =
-          pkgs.callPackage ./nix-index-wrapper.nix {
-            nix-index-database = databases.${pkgs.stdenv.system}.database;
-          };
-        comma-with-db =
-          pkgs.callPackage ./comma-wrapper.nix {
-            nix-index-database = databases.${pkgs.stdenv.system}.database;
-          };
-      };
+      testSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+      ];
+
+      systems = testSystems ++ [
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      mkPackages =
+        pkgs:
+        let
+          generated = import ./generated.nix;
+
+          nix-index-database =
+            (pkgs.fetchurl {
+              url = generated.url + pkgs.stdenv.system;
+              hash = generated.hashes.${pkgs.stdenv.system};
+            }).overrideAttrs
```

Comment:
`overrideAttrs` is the only way to do this because `passthru` doesn't work for some reason and `fetchurl` will error at being passed `__structedAttrs` or `unsafeDiscardReferences` `overrideAttrs` bypasses all of `fetchurl` argument checking and overrides the underlying `mkDerivation` call

Reply:
I think it's ok to use `unsafeDiscardReferences` but is there an easy way to detect if the feature is not present and bail out with a error message?